Source/WebCore/page/FocusController.cpp

```diff
@@ -481,16 +481,13 @@ void FocusController::setFocusedFrame(Frame* frame, BroadcastFocusedFrame broadc
     m_isChangingFocusedFrame = false;
 }
 
-LocalFrame& FocusController::focusedOrMainFrame() const
+RefPtr<LocalFrame> FocusController::focusedOrMainFrame() const
```

Comment:
This goes against our usual patterns. I think this should return a `LocalFrame*`, not a RefPtr. This doesn't transfer ownership and we don't want to cause unnecessary churn.

Reply:
cc @rniwa

Reply:
getters that return a RefPtr are named `protectedFoo()` and are only used to avoid local variables:
e.g.
```
focusController->protectedFocusedOrMainFrame()->doNonTrivialStuff();
```

We'd still use the `focusedOrMainFrame()` one that returns a raw pointer in those cases:
```
RefPtr frame = focusController->focusedOrMainFrame();
```
or
```
return focusController->focusedOrMainFrame()->trivialInlineGetter(); // To avoid unnecessary refcounting churn.
```

Reply:
Ok.  The last one would definitely need a null check.  That's the whole point of this PR.  Sometimes with site isolation we neither have a local main frame nor a focused frame.

Reply:
Yes, but is doesn't matter, the pointer can stay raw in that case.